R/logit_mle_newton.R

```diff
@@ -5,13 +5,35 @@ are_all_close <- function(v, w, abs_tol = 1e-6, rel_tol = 1e-6) {
   return(are_all_within_atol && are_all_within_rtol)
 }
 
-logit_newton = function(design, outcome, maxiter = 1000){
+take_one_newton_step = function(design, outcome, beta_old, newton_opt = "qr"){
+  hessian = logit_log_likelihood_hessian(beta_old, design, outcome)
+  gradient = logit_log_likelihood_gradient(beta_old, design, outcome)
+  supported_method <- c("qr", "lu", "chol")
+  if (!(newton_opt %in% supported_method)) {
+    stop("Inputed newton_opt is not supported, please use qr, lu or chol")
+  }
+  if (newton_opt == "lu") {
+    beta_new = beta_old - solve(hessian, gradient)
```

Comment:
```suggestion

  supported_method <- c("qr", "lu", "chol")
  if (!(newton_opt %in% supported_method)) {
    stop("Inputed newton_opt is not supported, please use qr, lu or chol")
  }
  if (newton_opt == "lu") {
    hessian = logit_log_likelihood_hessian(beta_old, design, outcome)
    gradient = logit_log_likelihood_gradient(beta_old, design, outcome)
    beta_new = beta_old - solve(hessian, gradient)
```

-1

If solver is not lu, you're unnecessarily computing hessian and grad